src/condor_utils/proc_family_direct_cgroup_v1.cpp

```diff
@@ -221,6 +223,26 @@ ProcFamilyDirectCgroupV1::cgroupify_process(const std::string &cgroup_name, pid_
 	// and save the eventfd, so we can read from it when the job exits to 
 	// check for ooms
 	cgroup_eventfd_map[pid] = efd;
+
+
+	// Remove devices we want to make unreadable.  Note they will still appear
+	// in /dev, but any access will fail, even if permission bits allow for it.
+	for (dev_t dev: this->cgroup_hide_devices) {
+		stdfs::path cgroup_device_deny_path = cgroup_root_dir / "devices" / cgroup_name / "devices.deny";
+		int devd = open(cgroup_device_deny_path.c_str(), O_WRONLY, 0666);
+		if (devd > 0) {
+			std::string deny_command;
+			formatstr(deny_command, "c %d:%d rwm", major(dev), minor(dev));
+			dprintf(D_ALWAYS, "Cgroupv1 hiding device with %s\n", deny_command.c_str());
```

Comment:
Just to confirm: this isn't a debugging D_ALWAYS?

Reply:
I intentionally made this D_ALWAYS, given the severity of what it is doing, and the frequency at which it will be done (once per starter invocation).

Reply:
Sounds good, thanks.